src/request/actions/ActionCreator.ts

```diff
@@ -0,0 +1,32 @@
+import { Action } from 'redux';
+
+interface IActionCreatorAction<TPayload> extends Action {
+    payload: TPayload;
+}
+
+interface IActionCreatorBase<TPayload> {
+    type: string;
+    unwrap(action: Action): TPayload;
+}
+
+interface IActionCreator<TPayload> extends IActionCreatorBase<TPayload> {
+    (payload: TPayload): IActionCreatorAction<TPayload>;
+}
+
+export function createActionCreator<TPayload>(type: string): IActionCreator<TPayload> {
+    const actionCreator = (payload: TPayload) => {
+        return {
+            type: type,
+            payload: payload
```

Comment:
The `redux` documentation recommends [`flux-standard-action`](https://github.com/acdlite/flux-standard-action) as the pattern for actions, which suggests the `payload` property instead of an "inherited" model.  This also tended to simplify "unwrapping" the actions in the reducers from a TypeScript-typed perspective.